plugin/src/main/groovy/com/novoda/staticanalysis/internal/TasksCompat.groovy

```diff
@@ -0,0 +1,44 @@
+package com.novoda.staticanalysis.internal
+
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.UnknownTaskException
+import org.gradle.api.tasks.TaskCollection
+import org.gradle.util.GradleVersion
+
+class TasksCompat {
+
+    private static boolean IS_GRADLE_MIN_49 = GradleVersion.current() >= GradleVersion.version("4.9")
+
+    static <T extends Task> Object maybeCreateTask(Project project, String name, Class<T> type, Action<? super T> configuration) {
+        if (IS_GRADLE_MIN_49) {
+            try {
+                def provider = project.tasks.named(name)
```

Comment:
Here is an ugly solution to lack of `maybeRegister` API in Gradle. More info: https://github.com/gradle/gradle/issues/6243

Using `maybeCreate` may not be the best practice after all. But I've looked at our implementation. Since we want to be flexible and adjust to configuration of other tools and plugins, it is not easy to get rid of it. It is really valuable for us. 

They don't have lazy alternative of `maybeCreate`. But they also don't have any `lazy` query method. `named` is the new way. It returns `TaskProvider` instead. But it throws exception when the task is not found.

So, long story short, this is the only way to do lazy and maybe together.